addons/hr_holidays/models/hr_leave_type.py

```diff
@@ -231,16 +231,18 @@ def get_employees_days(self, employee_ids, date=None):
             ('holiday_status_id', 'in', self.ids)
         ])
 
-        if not date:
-            date = fields.Date.context_today(self)
```

Comment:
If you create an allocation valid from the 1st Feb, and request a leave on the 22nd Jan, your fix will still show that Time Off Type, even tho you don't have a valid allocation at that time.

Instead, I'd leave how it was initially and just use
```py
            date = self.env.context.get('default_date_from', fields.Date.context_today(self))
```